### PR TITLE
UI: Use selected file path when opening file dialogs

### LINKED ISSUE #6777: OBS not saving last path when browsing for LUT filter
### Operating System Info

Windows 10

### Other OS

_No response_

### OBS Studio Version

27.2.4

### OBS Studio Version (Other)

_No response_

### OBS Studio Log URL

https://obsproject.com/logs/qYRCAtNpCPG4fZOM

### OBS Studio Crash Log URL

_No response_

### Expected Behavior

I expect that OBS saves the last folder I was viewing when I selected the filter I want to use, so I can try out multiple filters

### Current Behavior

When selecting a LUT filter for a source, it does not save the last path you were browsing at that moment, but it saves when you click "Close" and open the window up again.

### Steps to Reproduce

1. Add source like a webcam or something
2. Open Filters
3. Apply LUT Filter
4. Browse for a filter, and go to a folder that is not the default obs-filters folder
5. Select one
6. Browse for a filter again
7. The last path you were in did not save.

Now do this again but instead of browsing for a filter again, close the filters window, open it again then browse for a filter, it will then have your last path saved, but this is not the wanted and convenient behavior.
...

### Anything else we should know?

_No response_

### PR DESCRIPTION
**Pull request by Warchamp7**
### Description
Updates the properties view to use the existing path of a selected file when creating file dialogs, rather than relying on logic existing where `obs_properties_add_path` is called.

### Motivation and Context
Fixes #6777 at a deeper level compared to the approach in #6786

Currently, file dialogs for property views reference default_path for their working directory. However, this is set only when the property view is being created. Selecting a file or resetting to defaults does not update this value. Due to most places in the the code having a default_path of NULL, behaviour was _usually_ as expected, as re-opening the file dialog would simply fall back to the last location browsed (on Windows at least).

In the case of the LUT filter, we specify a default_path where the default LUT files are located. This meant that after selecting a LUT file outside that folder, immediately re-opening the file dialog would start in the LUT folder again, rather than the directory of the selected file. The default_path would only get updated to the selected files location when the Filters dialog is re-opened or the filters properties are reloaded via switching off of it and back.

Rather than making sure we update default_path anywhere a file selection can happen, update properties view itself to use the selected path when creating the file dialog, allowing default_path to act strictly as the fallback when there is nothing selected.

### How Has This Been Tested?
Opened the file selectors for the LUT filter, Image slideshow properties, Image source properties, and Image source context bar. Ensured that the file dialog always opened to the location of the specified file if set, the default_path if it's not NULL, and native OS fallback in the case of neither.

### Types of changes
- Bug fix (non-breaking change which fixes an issue)

- Tweak (non-breaking change to improve existing functionality)

- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Review by RytoEX — CHANGES_REQUESTED** [review]
The second commit should either use the prefix "plugins: " or be split into two commits, one for "image-source: " and one for "obs-filters: ".

File: UI/properties-view.cpp [hunk 1]
```diff
@@ -1812,18 +1812,23 @@ bool WidgetInfo::PathChanged(const char *setting)
 	const char *desc = obs_property_description(property);
 	obs_path_type type = obs_property_path_type(property);
 	const char *filter = obs_property_path_filter(property);
+	const char *last_path = obs_data_get_string(view->settings, setting);
```
Comment (derrod) [inline UI/properties-view.cpp:1815]: This seems unused.

**Review by derrod — CHANGES_REQUESTED** [review]

**Comment by norihiro** [comment]
The commit 7b72331f0 should have the prefix `plugins:` instead of `Plugins:`.

**Review by derrod — APPROVED** [review]
Seems fine aside from the commit message complaints.

**Pull request merged**